src/model/Position.ts

```diff
@@ -40,6 +44,34 @@ export class Position {
     feeGrowthInside0X128: JSBI,
     feeGrowthInside1X128: JSBI
   ) {
-    // TODO
+    let liquidityNext: JSBI;
+    if (JSBI.equal(liquidityDelta, ZERO)) {
+      assert(JSBI.lessThanOrEqual(this.liquidity, ZERO), "NP");
```

Comment:
Question here, we should use assert to make condition judgement or we also could use like `console.error()`

Reply:
Good question, the more optimized way is to `throw` directly, it has the least overhead.
However, here we are trying to mimic the format of a solidity smart contract, e.g. `require(...)`, that's why we are using the native `assert` library from CommonJS, ultimately it's doing a `throw` but definitely with more overhead.
Uniswap is using `tiny-invariant` which also does a `throw` but has very little overhead yet keeping the format similar to `require(...)`, we can switch to that if the overhead is unbearable.